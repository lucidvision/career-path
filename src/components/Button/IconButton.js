import React from 'react'
import PropTypes from 'prop-types'
import { StyleSheet, Text, TouchableOpacity } from 'react-native'

IconButton.propTypes = {
  onPress: PropTypes.func.isRequired,
  title: PropTypes.string.isRequired
}

IconButton.defaultProps = {
  onPress: () => {},
  title: ''
}

export default function IconButton (props) {
  return (
    <TouchableOpacity
      activeOpacity={0.8}
      style={styles.button}
      {...props}>
      {props.children}
      <Text>{props.title}</Text>
    </TouchableOpacity>
  )
}

const styles = StyleSheet.create({
  button: {
    alignItems: 'center',
    borderColor: 'black',
    borderRadius: 10,
    borderWidth: 2,
    elevation: 4,
    height: 100,
    justifyContent: 'center',
    padding: 10,
    shadowColor: 'rgba(79, 108, 141, 0.5)',
    shadowOffset: {
      height: 8
    },
    shadowOpacity: 0.8,
    shadowRadius: 6,
    width: 100
  }
})
