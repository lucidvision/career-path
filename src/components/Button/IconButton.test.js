import React from 'react'
import { shallow } from 'enzyme'
import IconButton from './IconButton'

const createIconButton = (override) => {
  const props = {
    onPress: () => {},
    title: '',
    ...override
  }

  return shallow(<IconButton {...props} />)
}

describe('IconButton', () => {
  const component = createIconButton()

  it('renders correctly', () => {
    expect(component).toMatchSnapshot()
  })

  describe('renders the correct output', () => {
    it('displays the correct text', () => {
      const component = createIconButton({title: 'Rainbow'})
      expect(component.find('Text').props().children).toBe('Rainbow')
    })

    it('triggers callback on press', () => {
      const mockOnPress = jest.fn()
      const component = createIconButton({onPress: mockOnPress})
      const touchableOpacity = component.find('TouchableOpacity')
      touchableOpacity.props().onPress()
      expect(mockOnPress).toHaveBeenCalled()
    })
  })
})
