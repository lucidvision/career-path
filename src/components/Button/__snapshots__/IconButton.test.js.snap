// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton renders correctly 1`] = `
<TouchableOpacity
  activeOpacity={0.8}
  onPress={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderColor": "black",
      "borderRadius": 10,
      "borderWidth": 2,
      "elevation": 4,
      "height": 100,
      "justifyContent": "center",
      "padding": 10,
      "shadowColor": "rgba(79, 108, 141, 0.5)",
      "shadowOffset": Object {
        "height": 8,
      },
      "shadowOpacity": 0.8,
      "shadowRadius": 6,
      "width": 100,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  />
</TouchableOpacity>
`;
